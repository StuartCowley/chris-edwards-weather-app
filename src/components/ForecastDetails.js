import React from "react";
import "../styles/ForecastDetails.css";
import {
  WaterOutline,
  SpeedometerOutline,
  CompassOutline,
  ThermometerOutline,
} from "react-ionicons";

function ForecastDetails({ forecast }) {
  return (
    <div className="forecast-details">
      <div className="forecast-detail">
        <div className="icon-container">
          <WaterOutline color="#fff" width="50px" height="50px" />
        </div>
        <p>Humidity: {forecast.humidity}%</p>
      </div>
      <div className="forecast-detail">
        <div className="icon-container">
          <SpeedometerOutline color="#fff" width="50px" height="50px" />
        </div>
        <p>Wind speed: {forecast.wind.speed}mph</p>
      </div>
      <div className="forecast-detail">
        <div className="icon-container">
          <CompassOutline color="#fff" width="50px" height="50px" />
        </div>
        <p>Wind direction: {forecast.wind.direction}</p>
      </div>
      <div className="forecast-detail">
        <div className="icon-container">
          <ThermometerOutline color="#fff" width="50px" height="50px" />
        </div>
        <p>
          Temp min/max: {forecast.temperature.min}°C /{" "}
          {forecast.temperature.max}
          °C
        </p>
      </div>
    </div>
  );
}

export default ForecastDetails;
