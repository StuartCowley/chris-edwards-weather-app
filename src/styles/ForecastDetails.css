.forecast-details {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  background-color: #475881;
  border-radius: 0.5rem;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.1);
  padding: 1rem;
  height: 200px;
  box-sizing: border-box;
}

.forecast-detail {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.icon-container {
  margin-bottom: 0.5rem;
}
