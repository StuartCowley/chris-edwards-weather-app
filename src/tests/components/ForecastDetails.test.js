import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    humidity: 71,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct forecast details", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText(`Humidity: ${validProps.humidity}%`)).toBeInTheDocument();
    expect(
      getByText(`Wind speed: ${validProps.wind.speed}mph`)
    ).toBeInTheDocument();
    expect(
      getByText(`Wind direction: ${validProps.wind.direction}`)
    ).toBeInTheDocument();
    expect(
      getByText(
        `Temp min/max: ${validProps.temperature.min}°C / ${validProps.temperature.max}°C`
      )
    ).toBeInTheDocument();
  });
});
