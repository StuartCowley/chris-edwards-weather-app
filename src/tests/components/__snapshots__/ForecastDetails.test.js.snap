// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-detail"
    >
      <div
        class="icon-container"
      >
        <span
          class="sc-beqWaB fHvFQc"
          color="#fff"
          height="50px"
          rotate="0"
          width="50px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 320c0 88.37-55.63 144-144 144s-144-55.63-144-144c0-94.83 103.23-222.85 134.89-259.88a12 12 0 0118.23 0C296.77 97.15 400 225.17 400 320z"
              fill="none"
              stroke="currentColor"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M344 328a72 72 0 01-72 72"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
          </svg>
        </span>
      </div>
      <p>
        Humidity: 71%
      </p>
    </div>
    <div
      class="forecast-detail"
    >
      <div
        class="icon-container"
      >
        <span
          class="sc-beqWaB fHvFQc"
          color="#fff"
          height="50px"
          rotate="0"
          width="50px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M326.1 231.9l-47.5 75.5a31 31 0 01-7 7 30.11 30.11 0 01-35-49l75.5-47.5a10.23 10.23 0 0111.7 0 10.06 10.06 0 012.3 14z"
            />
            <path
              d="M256 64C132.3 64 32 164.2 32 287.9a223.18 223.18 0 0056.3 148.5c1.1 1.2 2.1 2.4 3.2 3.5a25.19 25.19 0 0037.1-.1 173.13 173.13 0 01254.8 0 25.19 25.19 0 0037.1.1l3.2-3.5A223.18 223.18 0 00480 287.9C480 164.2 379.7 64 256 64z"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
            <path
              d="M256 128v32M416 288h-32M128 288H96M165.49 197.49l-22.63-22.63M346.51 197.49l22.63-22.63"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
          </svg>
        </span>
      </div>
      <p>
        Wind speed: 13mph
      </p>
    </div>
    <div
      class="forecast-detail"
    >
      <div
        class="icon-container"
      >
        <span
          class="sc-beqWaB fHvFQc"
          color="#fff"
          height="50px"
          rotate="0"
          width="50px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
              fill="none"
              stroke="currentColor"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M350.67 150.93l-117.2 46.88a64 64 0 00-35.66 35.66l-46.88 117.2a8 8 0 0010.4 10.4l117.2-46.88a64 64 0 0035.66-35.66l46.88-117.2a8 8 0 00-10.4-10.4zM256 280a24 24 0 1124-24 24 24 0 01-24 24z"
            />
          </svg>
        </span>
      </div>
      <p>
        Wind direction: s
      </p>
    </div>
    <div
      class="forecast-detail"
    >
      <div
        class="icon-container"
      >
        <span
          class="sc-beqWaB fHvFQc"
          color="#fff"
          height="50px"
          rotate="0"
          width="50px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M307.72 302.27a8 8 0 01-3.72-6.75V80a48 48 0 00-48-48h0a48 48 0 00-48 48v215.52a8 8 0 01-3.71 6.74 97.51 97.51 0 00-44.19 86.07A96 96 0 00352 384a97.49 97.49 0 00-44.28-81.73zM256 112v272"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <circle
              cx="256"
              cy="384"
              r="48"
            />
          </svg>
        </span>
      </div>
      <p>
        Temp min/max: 12°C / 22°C
      </p>
    </div>
  </div>
</DocumentFragment>
`;
